### PR TITLE
add tolerance parm and example

### PR DESCRIPTION
**Pull request by akaszynski**
This PR adds the `tolerance` parameter in response to https://github.com/pyvista/pyvista-support/issues/480

It also adds a basic example demonstrating the use of the method.

### CODE REVIEW

**Comment by akaszynski** [comment]
Replaces #1595

File: pyvista/core/filters/composite.py [hunk 1]
```diff
@@ -21,21 +21,43 @@ def extract_geometry(composite):
...
-        """Append all blocks into a single unstructured grid.
+    def combine(composite, merge_points=False, tolerance=0.0):
+        """Combine all blocks into a single unstructured grid.
 
         Parameters
         ----------
         merge_points : bool, optional
             Merge coincidental points.
 
+        tolerance : float
```
Comment (adeak) [inline pyvista/core/filters/composite.py:32]:
```suggestion
        tolerance : float, optional
```

File: pyvista/core/filters/composite.py [hunk 2]
```diff
@@ -21,21 +21,43 @@ def extract_geometry(composite):
...
+            The tolerance to use to find coincident points when
```
Comment (adeak) [inline pyvista/core/filters/composite.py:33]: This seems to be the default, looking at https://vtk.org/doc/nightly/html/classvtkAppendFilter.html#a79f99dabfc8fa2b1a880ab0c07955226. Do we want to expose the absolute/relative choice? Somehow the "relative to mesh diagonal" seems rarely useful to me.

```suggestion
            The absolute tolerance to use to find coincident points when
```

Comment (akaszynski) [reply pyvista/core/filters/composite.py:33]: I don't see relative tolerance to be that helpful, so let's no expose it.

File: pyvista/core/filters/composite.py [hunk 3]
```diff
@@ -21,21 +21,43 @@ def extract_geometry(composite):
...
+
+        >>> merged = block.combine(merge_points=True)
+        >>> merged.n_points
+        12
+
         """
         alg = _vtk.vtkAppendFilter()
         for block in composite:
             if isinstance(block, _vtk.vtkMultiBlockDataSet):
                 block = CompositeFilters.combine(block, merge_points=merge_points)
```
Comment (adeak) [inline pyvista/core/filters/composite.py:57]: Almost missed this!
```suggestion
                block = CompositeFilters.combine(block, merge_points=merge_points,
                                                 tolerance=tolerance)
```

**Review by adeak — APPROVED** [review]
LGTM with one change I'm going to commit right now.

**Pull request merged**